ydb/library/yql/dq/runtime/dq_output_consumer.cpp

```diff
@@ -391,22 +388,14 @@ class TDqOutputHashPartitionConsumerBlock : public IDqOutputConsumer {
             outputBlockIndexes[GetHashPartitionIndex(datums.data(), i)].push_back(i);
         }
 
-        ui64 maxLen = 0;
-        for (auto& indexes : outputBlockIndexes) {
-            maxLen = std::max(maxLen, indexes.size());
-        }
-
-        if (maxLen > MaxOutputBlockLen_) {
-            MakeBuilders(maxLen);
-        }
-
         TVector<std::unique_ptr<TArgsDechunker>> outputData;
         for (size_t i = 0; i < Outputs_.size(); ++i) {
             ui64 outputBlockLen = outputBlockIndexes[i].size();
             if (!outputBlockLen) {
                 outputData.emplace_back();
                 continue;
             }
+            MakeBuilders(outputBlockLen);
```

Comment:
Since we always recreate Builders, we may use Builders_[j]->Build(true) below to avoid extra mem allocation after Build() call

Reply:
Fixed, thank you!